modules/features/player/src/main/java/au/com/shiftyjelly/pocketcasts/player/view/BookmarksFragment.kt

```diff
@@ -255,6 +287,112 @@ private fun PlayButton(
     }
 }
 
+@Composable
+private fun NoBookmarksView() {
+    MessageView(
+        titleView = {
+            TextH20(
+                text = stringResource(LR.string.bookmarks_not_found),
+                color = MaterialTheme.theme.colors.playerContrast01,
+            )
+        },
+        buttonTitleRes = LR.string.bookmarks_headphone_settings,
+        buttonAction = { /* TODO */ },
+    )
+}
+
+@Composable
+private fun PlusUpsellView() {
+    MessageView(
+        titleView = {
+            Image(
+                painter = painterResource(IR.drawable.pocket_casts_plus_logo),
+                contentDescription = stringResource(LR.string.pocket_casts),
+            )
+        },
+        buttonTitleRes = LR.string.subscribe, // TODO: Bookmarks update upsell button title based on subscription status
+        buttonAction = { /* TODO */ },
+    )
+}
+
+@Composable
+private fun MessageView(
+    titleView: @Composable () -> Unit = {},
+    @StringRes buttonTitleRes: Int,
+    buttonAction: () -> Unit = {},
+) {
+    Column(
+        modifier = Modifier
+            .fillMaxHeight()
+            .verticalScroll(rememberScrollState())
+    ) {
+        Box(
+            modifier = Modifier
+                .padding(16.dp)
+                .background(
+                    color = MaterialTheme.theme.colors.playerContrast06,
+                    shape = RoundedCornerShape(size = 4.dp)
+                )
+        ) {
+            Column(
+                verticalArrangement = Arrangement.Center,
+                horizontalAlignment = Alignment.CenterHorizontally,
+                modifier = Modifier
+                    .padding(start = 16.dp, end = 16.dp, top = 24.dp, bottom = 16.dp)
+            ) {
+                titleView()
+                TextP40(
+                    text = stringResource(LR.string.bookmarks_create_instructions),
+                    color = MaterialTheme.theme.colors.playerContrast02,
+                    textAlign = TextAlign.Center,
+                    modifier = Modifier.padding(top = 12.dp)
+                )
+                TextButton(
+                    buttonTitleRes = buttonTitleRes,
+                    buttonAction = buttonAction,
+                    modifier = Modifier.padding(top = 4.dp)
+                )
+            }
+        }
+    }
+}
+
+@Composable
+private fun TextButton(
+    @StringRes buttonTitleRes: Int,
+    buttonAction: () -> Unit = {},
+    modifier: Modifier,
+) {
+    Box(
+        contentAlignment = Alignment.Center,
+        modifier = modifier
+            .minimumInteractiveComponentSize()
```

Comment:
Thanks again for the Compose lesson! I didn't know about this method.

It would be great to make this component reusable. Maybe not as part of this pull request, but maybe we can try [Showkase](https://github.com/airbnb/Showkase) to make it easier to find and share components.